app/src/main/java/com/example/bookwormadventuresdeluxe2/SetLocationActivity.java

```diff
@@ -1,42 +1,599 @@
+                            .create()
+                            .show();
+                }
+
+
+            }
+            else
+            {
+                checkSettingsAndStartLocationUpdates();
+            }
+            mapFragment.getMapAsync(this);        /* Permissions were requested, set On Map ready Callback */
+        }
+    }
+
+    /**
+     * Process when map ready
+     *
+     * @param googleMap
+     */
     @Override
     public void onMapReady(GoogleMap googleMap)
     {
         map = googleMap;
+        map.setMapType(GoogleMap.MAP_TYPE_NORMAL);
+
+        map.setOnMapClickListener(this); // Listener to set marker
+        map.setOnMarkerDragListener(this);// Listener to drag and move marker
+
+        /* Enable user location if permission granted*/
+        if (ActivityCompat.checkSelfPermission(this,
+                Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED
+                && ActivityCompat.checkSelfPermission(this,
+                Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_GRANTED)
+        {
+
+            enableUserLocation();
+        }
+
+        /* Move Camera to default location */
+        LatLng latLng = new LatLng(defaultLatitude, defaultLongitude);
+        map.animateCamera(CameraUpdateFactory.newLatLngZoom(latLng, CAMERA_ZOOM_LEVEL));
+
+    }
+
+    /**
+     * Enables  User Location on Map
+     * Ok to SuppressLint as permission checked before call to mehod
+     */
+    @SuppressLint("MissingPermission")
+    private void enableUserLocation()
+    {
+
+        if (map != null)
+        {
+            map.setMyLocationEnabled(true);
+            map.setPadding(0, 500, 0, 0); // Move Get My location button under TextBox
+        }
+
+    }
+
+    /**
+     * Returns a list of permissions not granted from a list of permissions passed in
+     *
+     * @param wantedPermissions the list of permssions required
+     * @return list of permissions requiring approval
+     */
+    private ArrayList<String> permissionsToRequest(ArrayList<String> wantedPermissions)
+    {
+        /* List of permissions not granted */
+        ArrayList<String> result = new ArrayList<>();
+
+        for (String perm : wantedPermissions)
+        {
+            if (!hasPermission(perm))
+            {
+                result.add(perm);
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Can check if permission was previously accepted on SDK > M
+     * On sdk < M user must have accepted before download
+     */
+    private boolean hasPermission(String perm)
+    {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
+        {
+            return checkSelfPermission(perm) == PackageManager.PERMISSION_GRANTED;
+        }
+        return true;
+    }
+
+    /**
+     * Add new marker on Map click
+     *
+     * @param latLng map click location
+     */
+    @Override
+    public void onMapClick(LatLng latLng)
+
```

Comment:
nit weird space

Reply:
Thanks for noticing. I checked other places for redundant spacing.